### ISSUE TITLE
Accessibility Story: Upgrade "skip to main" links on all pages: 2

### ISSUE THREAD

**Issue by paregorios** [isawnyu/isaw.web#231]
**As a user with extreme low vision or blindness, I will encounter a "skip to main" link on each page that makes it easy for me to bypass header and other ancillary material to get directly to the main content area of that page.**

Specific considerations, per Microassist:

> - role = complementary (ARIA), aside (HTML5)
> - As there are more than four elements between the top of the page and the main content area, a skip to main link will be required for mobility AT users.
> - Should have aria-label of “skip to main”

W3 Guidance: [Techniques for WCAG 2.0 > G1: Adding a link at the top of each page that goes directly to the main content area](https://www.w3.org/TR/2016/NOTE-WCAG20-TECHS-20161007/G1)

 - link is the first focusable control on the Web page
 - description of the link communicates that it links to the main content
 - link is either always visible or visible when it has keyboard focus (but see below).
 - activating the link moves the focus to the main content
 - after activating the link, the keyboard focus remains on the main content

> Note: It is preferable for links to be visible at all times, since users navigating via the keyboard include switch users, those using techniques that generate keyboard strokes slowly, screen magnification software users, screen reader users working with sighted colleagues, keyboard only users and those navigating using voice recognition software. However, Success Criterion 2.4.1 does not require that they be visible when they do not have focus, and links that are visible only when they have focus can meet this success criterion.

Traceability: 

- [Microassist Report 1 > Home Page Template > Landmarks](https://docs.google.com/document/d/10cfUFx-o_kSGBFKCe4MtL45cyc-c-Gwaz3Dxlrp8Qgg/edit#bookmark=id.7rvkgwq4gpnr) (and so on all following "template" sections in the report)
- [Microassist Report 1 > Tab Order](https://docs.google.com/document/d/10cfUFx-o_kSGBFKCe4MtL45cyc-c-Gwaz3Dxlrp8Qgg/edit#bookmark=id.rb00k1yddy62)
- [Microassist Report 1 > WCAG Compliance > Bypass Blocks](https://docs.google.com/document/d/10cfUFx-o_kSGBFKCe4MtL45cyc-c-Gwaz3Dxlrp8Qgg/edit#bookmark=id.u30lkvd2c40g)

**Comment by paregorios** [comment]
Per [Microassist Report 1 > WCAG Compliance > Bypass Blocks](https://docs.google.com/document/d/10cfUFx-o_kSGBFKCe4MtL45cyc-c-Gwaz3Dxlrp8Qgg/edit#bookmark=id.u30lkvd2c40g):

> NYU Institute for the Study of the Ancient World currently has a skip to main and skip to navigation, but they are not visible. Skip to main links should be visible and activating the link should set focus beyond the header content to the main content.

**Comment by paregorios** [comment]
This is probably the template that provides this functionality everywhere on the site and therefore is presumably where we would want to add attributes etc. 

https://github.com/isawnyu/isaw.web/blob/master/src/isaw.theme/isaw/theme/browser/template-overrides/plone.app.layout.viewlets.skip_links.pt

The site theme css uses the "hiddenStructure" class to turn off display of this section. To comply with the requirement that it be visible (at least when focused), we will modify the CSS. I would think that, since we are no longer "hiding" this structure, we probably want to change the class value to something more semantically useful, like "bypass blocks".

**Comment by paregorios** [comment]
NB #69

**Comment by paregorios** [comment]
This issue is now in work in branch [issue231](https://github.com/isawnyu/isaw.web/tree/issue231).

**Comment by paregorios** [comment]
https://www.w3.org/TR/wai-aria-1.1/#aria-label

> If the label text is visible on screen, authors SHOULD use aria-labelledby and SHOULD NOT use aria-label. There may be instances where the name of an element cannot be determined programmatically from the content of the element, and there are cases where providing a visible label is not the desired user experience. Most host languages provide an attribute that could be used to name the element (e.g., the title attribute in HTML), yet this could present a browser tooltip. In the cases where a visible label or visible tooltip is undesirable, authors MAY set the accessible name of the element using aria-label. As required by the text alternative computation, user agents give precedence to aria-labelledby over aria-label when computing the accessible name property.

**Comment by paregorios** [comment]
https://docs.plone.org/4/en/old-reference-manuals/plone_3_theming/elements/hiddenelements/plone.skip_links.html

**Comment by paregorios** [comment]
We want to see this:

## HTML changes

Right now at the top of each page one finds the following markup:

```html
<p class="hiddenStructure">Navigation</p>
<p class="hiddenStructure">
  <a accesskey="2" href="#content">Skip to content.</a> |
  <a accesskey="6" href="#main-navigation">Skip to navigation</a>
</p>
```

Make the following changes

 1. Eliminate "navigation" paragraph
 1. Change ```<p class="hiddenStructure">``` to ```<aside id="skip">```
 1. Eliminate "skip to navigation" link
 1. Change text "skip to content" to "skip to main content" 
 1. add ```aria-label = "skip to main"``` to the ```<a>``` tag
 
We should end up with:

```html
<aside id="skip">
  <a accesskey="2" href="#content" aria-label="skip to main">Skip to main content</a>
</aside>
```
### CSS changes will also be required in order the achieve the following changes:

```css
#skip {
    height: 1rem;
    line-height: 1rem;
    vertical-align: middle;
    padding: 4px;
    font-size: 80%;
    background-color: #fffaf8;
}
```

Also: style ```#skip > a``` so it behaves just like links in body text (i.e., as specified in #353)

## Result

It should look like this:

![screen shot 2018-12-05 at 5 07 20 pm](https://user-images.githubusercontent.com/263285/49550098-4330bc00-f8b0-11e8-84a7-829b02efb9e4.png)

**Comment by paregorios** [comment]
The "skip to main content" link is where we want it and is structured the way we want it. It still needs a couple of style tweaks so that it has the same appearance as in-body links for: default/visited (red brick solid), focus/hover (blue dashed -- this works), active (halloween).

**Comment by alecpm** [comment]
Should be styled consistently now.

**Comment by paregorios** [comment]
Hmmm. I'm still not seeing this on staging with any of my three browsers, even after clearing cache and reloading. Could old CSS be cached server-side?

**Comment by alecpm** [comment]
It could have been. Please try again, I'm definitely seeing the styling on staging.

**Comment by paregorios** [comment]
After setting portal_css to debug mode in the ZMI, I can now see the new styling on staging. I have set portal_css back to not debug. Looks good. Please merge to master and deploy to production.

**Comment by paregorios** [comment]
This has clearly been deployed to production. I assume it has also been merged to master. Closing (@alecpm please confirm it has been merged to master so we don't discover a downstream regression). Thanks.

**Comment by alecpm** [comment]
This has indeed been merged to master and deployed to production.
